server.go

```diff
@@ -2664,6 +2677,143 @@ func (s *Server) executeShowStatsStatement(stmt *influxql.ShowStatsStatement, us
 	return &Result{Series: rows}
 }
 
+func (s *Server) executeShowDiagnosticsStatement(stmt *influxql.ShowDiagnosticsStatement, user *User) *Result {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	rows := make([]*influxql.Row, 0)
+	now := time.Now().UTC().Format(time.RFC3339Nano)
+
+	var m runtime.MemStats
+	runtime.ReadMemStats(&m)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
+
+	diags := []struct {
+		name   string
+		fields map[string]interface{}
+	}{
+		{
+			name: "build",
+			fields: map[string]interface{}{
+				"version":    s.Version,
+				"commitHash": s.CommitHash,
+			},
+		},
+		{
+			name: "server",
+			fields: map[string]interface{}{
+				"id":                  s.id,
+				"path":                s.path,
+				"authEnabled":         s.authenticationEnabled,
+				"index":               s.index,
+				"retentionAutoCreate": s.RetentionAutoCreate,
+				"numShards":           len(s.shards),
+			},
+		},
+		{
+			name: "cq",
+			fields: map[string]interface{}{
+				"lastRun": s.lastContinuousQueryRun,
+			},
+		},
+		{
+			name: "system",
+			fields: map[string]interface{}{
+				"startTime": startTime,
+				"uptime":    time.Since(startTime).String(),
+				"hostname":  hostname,
+				"pid":       os.Getpid(),
+				"os":        runtime.GOOS,
+				"arch":      runtime.GOARCH,
+				"numcpu":    runtime.NumCPU(),
+			},
+		},
+		{
+			name: "memory",
+			fields: map[string]interface{}{
+				"alloc":        m.Alloc,
+				"totalAlloc":   m.TotalAlloc,
+				"sys":          m.Sys,
+				"lookups":      m.Lookups,
+				"mallocs":      m.Mallocs,
+				"frees":        m.Frees,
+				"heapAlloc":    m.HeapAlloc,
+				"heapSys":      m.HeapSys,
+				"heapIdle":     m.HeapIdle,
+				"heapInUse":    m.HeapInuse,
+				"heapReleased": m.HeapReleased,
+				"heapObjects":  m.HeapObjects,
+				"pauseTotalNs": m.PauseTotalNs,
+				"numGC":        m.NumGC,
+			},
+		},
+		{
+			name: "go",
+			fields: map[string]interface{}{
+				"goMaxProcs":   runtime.GOMAXPROCS(0),
+				"numGoroutine": runtime.NumGoroutine(),
+				"version":      runtime.Version(),
+			},
+		},
+	}
```

Comment:
I think it'd be easier to use this data from a client or another system if it were a type:

```

type Diagnostics struct {
	Build struct {
		Version    string `json:"version"`
		CommitHash string `json:"commitHash"`
	} `json:"build"`

	Server struct {
		ID          string `json:"id"`
		Path        string `json:"path"`
		AuthEnabled bool   `json:"authenticationEnabled"`
		...
	}
}
```

And then you can get the current `Diagnostics` type using `NewDiagnostics()`. That would shorten this function quite a bit.